api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,326 +16,389 @@
-                AclPermission.WRITE.getMask(), true);
-        aclTestDao.createAclEntry(registryAclEntry2);
-
-        AclTestDao.AclEntry registryAclEntry3 = new AclTestDao.AclEntry(registryIdentity3, 1, adminSid,
-                AclPermission.WRITE.getMask(), true);
-        aclTestDao.createAclEntry(registryAclEntry3);
-        registryAclEntry3.setSid(testUserSid);
-        registryAclEntry3.setOrder(2);
-        aclTestDao.createAclEntry(registryAclEntry3);
+    @Autowired
+    private UserManager mockUserManager;
 
-        MockitoAnnotations.initMocks(this);
+    @InjectMocks
+    private final DockerRegistryManager registryManager = new DockerRegistryManager();
 
-        TestUtils.configureDockerClientMock(dockerClient, dockerClientFactoryMock);
+    private DockerRegistry registry1;
+    private DockerRegistry registry2;
+    private DockerRegistry registry3;
+    private ToolGroup toolGroup1;
+    private ToolGroup toolGroup2;
+    private Tool tool1;
+    private Tool tool2;
+    private Tool tool3;
 
-        dockerClientFactoryMock.setObjectMapper(new ObjectMapper());
+    @Before
+    public void setUp() {
+        registry1 = getDockerRegistry(ID, REPO, SIMPLE_USER, "");
+        registry2 = getDockerRegistry(ID_2, REPO_EXTERNAL_PATH, EXTERNAL_PATH, ADMIN_ROLE);
+        registry3 = getDockerRegistry(ID_3, REPO_NO_WRITE_ACCESS, ADMIN_ROLE, "");
 
-        when(dockerClientFactoryMock.getDockerClient(any(DockerRegistry.class)))
-                .thenReturn(dockerClient);
-        when(dockerClient.getImageTags(any(String.class), any(String.class)))
-                .thenReturn(Collections.singletonList("latest"));
+        toolGroup1 = getToolGroup(ID, "library", registry1.getId(), SIMPLE_USER);
+        toolGroup2 = getToolGroup(null, "library2", registry1.getId(), SIMPLE_USER);
+        ToolGroup toolGroup3 = getToolGroup(null, "library", registry2.getId(), SIMPLE_USER);
 
-        ToolVersion toolVersion = new ToolVersion();
-        toolVersion.setDigest("test_digest");
-        toolVersion.setSize(DOCKER_SIZE);
-        toolVersion.setVersion("test_version");
-        Mockito.doReturn(toolVersion).when(dockerClient)
-                .getVersionAttributes(Mockito.any(), Mockito.any(), Mockito.any());
+        tool1 = getTool(toolGroup1, IMAGE, registry1, SIMPLE_USER);
+        tool2 = getTool(toolGroup2, IMAGE_NEW_GROUP, registry1, SIMPLE_USER);
+        tool3 = getTool(toolGroup3, IMAGE, registry2, SIMPLE_USER);
 
-        doNothing().when(toolVersionManager).updateOrCreateToolVersion(anyLong(), anyString(), anyString(),
-                any(DockerRegistry.class), any(DockerClient.class));
+        mockUserContext();
+        MockitoAnnotations.initMocks(this);
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
-    @WithMockUser(username = ADMIN, roles = ADMIN_ROLE)
     public void testEventCanEnableToolInExistingGroupWithRightUserAccess() {
```

Comment:
Let's rename all tests to smth like `should....`, for example `shouldEnableToolInExistingGroupWithRightUserAccess`

Reply:
Done!